modules/pixi/styles.js

```diff
@@ -30,568 +31,529 @@ import { osmPavedTags } from '../osm/tags';
-  },
-  footway: {
-    casing: { width: 5, color: 0xffffff },
-    stroke: { width: 3, color: 0x998888, dash: [6, 6], cap: PIXI.LINE_CAP.BUTT }
-  },
-  crossing_marked: {
-    casing: { width: 5, color: 0xddccaa },
-    stroke: { width: 3, color: 0x4c4444, dash: [6, 3], cap: PIXI.LINE_CAP.BUTT }
-  },
-  crossing_unmarked: {
-    casing: { width: 5, color: 0xddccaa },
-    stroke: { width: 3, color: 0x776a6a, dash: [6, 4], cap: PIXI.LINE_CAP.BUTT }
-  },
-  cycleway: {
-    casing: { width: 5, color: 0xffffff },
-    stroke: { width: 3, color: 0x58a9ed, dash: [6, 6], cap: PIXI.LINE_CAP.BUTT }
-  },
-  bridleway: {
-    casing: { width: 5, color: 0xffffff },
-    stroke: { width: 3, color: 0xe06d5f, dash: [6, 6], cap: PIXI.LINE_CAP.BUTT }
-  },
-  corridor: {
-    casing: { width: 5, color: 0xffffff },
-    stroke: { width: 3, color: 0x8cd05f, dash: [2, 8], cap: PIXI.LINE_CAP.ROUND }
-  },
-  steps: {
-    casing: { width: 5, color: 0xffffff },
-    stroke: { width: 3, color: 0x81d25c, dash: [3, 3], cap: PIXI.LINE_CAP.BUTT }
-  },
-
-  river: {
-    casing: { width: 10, color: 0x444444 },
-    stroke: { width: 8, color: 0x77dddd }
-  },
-  stream: {
-    casing: { width: 7, color: 0x444444 },
-    stroke: { width: 5, color: 0x77dddd }
-  },
-  ridge: {
-    stroke: { width: 2, color: 0x8cd05f}  // rgb(140, 208, 95)
-  },
-
-  runway: {
-    casing: { width: 10, color: 0x000000, cap: PIXI.LINE_CAP.BUTT },
-    stroke: { width: 8, color: 0xffffff, dash: [24, 48], cap: PIXI.LINE_CAP.BUTT }
-  },
-  taxiway: {
-    casing: { width: 7, color: 0x444444 },
-    stroke: { width: 5, color: 0xffff00 }
-  },
-
-  railway: {
-    casing: { width: 7, color: 0x555555, cap: PIXI.LINE_CAP.BUTT },
-    stroke: { width: 2, color: 0xeeeeee, dash: [12, 12], cap: PIXI.LINE_CAP.BUTT,  }
-  },
-
-  ferry: {
-    casing: { alpha: 0 },  // disable
-    stroke: { width: 3, color: 0x58a9ed, dash: [12, 8], cap: PIXI.LINE_CAP.BUTT }
-  },
-
-  boundary: {
-    casing: { width: 6, color: 0x82b5fe, cap: PIXI.LINE_CAP.BUTT },
-    stroke: { width: 2, color: 0xffffff, dash: [20, 5, 5, 5], cap: PIXI.LINE_CAP.BUTT }
-  },
-  boundary_park: {
-    casing: { width: 6, color: 0x82b5fe, cap: PIXI.LINE_CAP.BUTT },
-    stroke: { width: 2, color: 0xb0e298, dash: [20, 5, 5, 5], cap: PIXI.LINE_CAP.BUTT }
-  },
-
-  barrier: {
-    casing: { alpha: 0 },  // disable
-    stroke: { width: 3, color: 0xdddddd, dash: [10, 5, 1, 5], cap: PIXI.LINE_CAP.ROUND }
-  },
-  barrier_wall: {
-    casing: { alpha: 0 },  // disable
-    stroke: { width: 3, color: 0xdddddd, dash: [10, 5, 1, 5], cap: PIXI.LINE_CAP.ROUND }
-  },
-  barrier_hedge: {
-    fill:   { color: 0x8cd05f, alpha: 0.3 },   // rgb(140, 208, 95)
-    casing: { alpha: 0 },  // disable
-    stroke: { width: 3, color: 0x8cd05f, dash: [10, 5, 1, 5], cap: PIXI.LINE_CAP.ROUND }
-  },
-
-  tree_row: {
-    casing: { width: 7, color: 0x444444 },
-    stroke: { width: 5, color: 0x8cd05f }
+export class STYLES {
+  constructor(context) {
```

Comment:
@Bonkles Alright, here is my first attempt at refactoring the styles.js file and updating the imports. 
> [!NOTE]
> I refactored the first object (motorway, trunk, primary ...) into an object called WAYS - that *may* be causing some unknown issues

But I am getting a TypeError now:

<img width="522" alt="Screenshot 2023-11-02 at 2 13 02 PM" src="https://github.com/facebook/Rapid/assets/95086147/76f5e7c0-2204-4e9f-a05c-e51c6f21a639">

Why can't I call the method from the import? 😢

Reply:
Ah! You will need to record the styleMatch method as a `static` method just to get it to be recognized by the system. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static

`static` methods are methods that do not require an `instance` of the class to run.

so, as a static method you should be able to call 

STYLES.styleMatch(<params>)


But as a regular method you'd have to do: 

let Styles = new STYLES(context); // Create a new instance of the class

Styles.styleMatch(<params) // call the method on that particular instance of the class

Reply:
Ahh, thank you.